mu/logic.py

```diff
@@ -234,6 +234,56 @@ def check_pycodestyle(code):
     return style_feedback
 
 
+def get_runtime_hex_path():
+    """
+    Returns the path to the hex runtime file - if this has been
+    specified under element 'microbit_runtime_hex' in settings.json.
+    This can be a fully-qualified file path, or just a file name
+    in which case the file should be located in the workspace directory.
+    """
+    runtime_hex_path = ""
+    settings_path = get_settings_path()
+    settings = {}
+    try:
+        with open(settings_path) as f:
+            settings = json.load(f)
+    except FileNotFoundError:
+        logger.error('Settings file {} does not exist.'.format(settings_path))
+    except ValueError:
+        logger.error('Settings file {} could not be parsed.'.format(
+                     settings_path))
+    else:
+        if 'microbit_runtime_hex' in settings and \
+                settings['microbit_runtime_hex'] != "":
+            runtime_hex_path = os.path.join(
```

Comment:
Don't make the mistake i did implementing `get_workspace_dir`!

Check the path exists (and is a file) here rather than later then this function will always return a *safe* value. (i.e. a valid path or nothing)

Regardless of whether or not my PR gets merged in uflash it would probably be better to return `None` instead of an empty string & check that the setting wasn't set to "" (more pythonic and all that).